src/schema.ts

```diff
@@ -147,79 +89,79 @@ export interface VibraphoneDropEvent {
+	 * The title of this program. It can be distinct from the title of the performance.
 	 */
-	readonly initialState: State;
+	title: string;
 	/**
-	 * The events that make up this performance. As with [[Program.dropEvents]],
-	 * these events must be in ascending order.
+	 * The author of this program. It can be distinct from the title of the performance.
 	 */
-	events: TimedEvent[];
+	author: string;
+	/**
+	 * Pulses per quarter.
+	 * This basically represents how many distinct pieces a quarter note is divisible by.
+	 * The higher the PPQ, the more flexibility when it comes to placing notes.
+	 */
+	readonly tpq: 240;
+	/** Version of VMMX in which the current program was made */
+	readonly version: string;
+	/** Total ticks on the "programming wheel" 240 PPQ \* 4 beats/measure \* 16 bars on wheel */
+	readonly length: 61440;
+	/** The amount of procrastination that occurred during the making of this program. */
+	procrastination?: number;
 }
-/** Metadata for performance */
-export interface PerformanceMetadata {
+
+/**
+ * Represents a program on the MMX's programming wheel. Used to store the programmed part of a performance.
+ */
+export interface Program {
 	/**
-	 * The title of the performance.
-	 * This doesn't have to be the same as the title of the program.
+	 * Information about this program. It can be distinct from a performance's metadata.
 	 */
-	title: string;
+	metadata: ProgramMetadata;
 	/**
-	 * The author/performer of the performance.
-	 * This doesn't have to be the same as the author of the program.
+	 * This is the "working" state of the machine, outside of a performance.
+	 * When a user is modifying the program, this is how we store what settings they use.
+	 * Whenever the user saves, this field gets updated, and whether they load, this gets read.
+	 * It has almost no relevance when editing a performance since the performance events will
+	 * cause it to be updated dynamically.
+	 *
+	 * ## Example
+	 * ```typescript
+	 * let prog: Program;
+	 * // invert the muted state of the snare
+	 * prog.state.machine.mute.snare = !prog.state.machine.mute.snare;
+	 * ```
 	 */
-	author: string;
+	state: State;
+	/**
+	 * All the [[TickedDropEvent]]s that occur during this program. They *must* be in ascending order by tick.
+	 */
+	dropEvents: TickedDropEvent[];
 }
 
-/**
- * An untimed event that changes the state of one channel
- * or manually drops a marble.
- *
- * Events only change one thing. Multiple simultaneous changes must
- * be represented by multiple events with the same tick.
- */
-export type Event =
-	| PerformanceDropEvent
-	| MachineEvent
-	| VibraphoneEvent
-	| HihatMachineEvent
-	| HihatEvent
-	| BassEvent;
+//#endregion
 
-/** Represents an event occurring in time (non-tempo dependent). */
-export type TimedEvent = CoreTimedEvent & Event;
-/** Represents information associated with all timed events. */
-export interface CoreTimedEvent {
-	/** The time (in seconds) that this event occurs. */
```

Comment:
Seconds? That's the most precision we can get for timed events?

Reply:
Seconds are floating point values, what do you mean?

Reply:
I was thinking in whole numbers. I think I see more milliseconds or nanoseconds in my programming which are whole numbers.